src/golang/cmd/server/handler/preview.go

```diff
@@ -234,20 +235,39 @@ func (h *PreviewHandler) Perform(ctx context.Context, interfaceArgs interface{})
 func (h *PreviewHandler) setupExecEnv(
 	ctx context.Context,
 	args *previewArgs,
-) (map[uuid.UUID]exec_env.ExecutionEnvironment, error) {
-	condaIntegration, err := exec_env.GetCondaIntegration(
+) (map[uuid.UUID]exec_env.ExecutionEnvironment, int, error) {
+	condaIntegration, condaConnectionState, err := exec_env.CondaConnectionState(
 		ctx, args.Id, h.IntegrationReader, h.Database,
 	)
 
 	dagSummary := args.DagSummary
 
 	if err != nil {
-		return nil, errors.Wrap(err, "Unable to verify if conda is connected.")
+		return nil, http.StatusInternalServerError, errors.Wrap(err, "Unable to verify if conda is connected.")
 	}
 
 	// For now, do nothing if conda is not connected.
-	if condaIntegration == nil {
-		return nil, nil
+	if condaConnectionState == nil {
+		return nil, http.StatusOK, nil
+	}
+
+	if condaConnectionState.Status == shared.FailedExecutionStatus {
```

Comment:
If this happens, what should the user do to fix?

Reply:
I think we can add a prompt in UI to ask user to delete the integration and reconnect. Ideally we could add a button to retry but there will be some work involved.

Reply:
add a prompt in UI sounds fine to me.